contracts/contracts/governance/Governor.sol

```diff
@@ -0,0 +1,159 @@
+        guardian = guardian_;
+    }
+
+    function proposeAndQueue(address[] memory targets, uint[] memory values, string[] memory signatures, bytes[] memory calldatas, string memory description) public returns (uint) {
+        require(msg.sender == guardian, "Governor::proposeAndQueue: sender must be gov guardian");
+        require(targets.length == values.length && targets.length == signatures.length && targets.length == calldatas.length, "Governor::propose: proposal function information arity mismatch");
+        require(targets.length != 0, "Governor::propose: must provide actions");
+        require(targets.length <= MAX_OPERATIONS, "Governor::propose: too many actions");
+
+        proposalCount++;
+        Proposal memory newProposal = Proposal({
+            id: proposalCount,
+            proposer: msg.sender,
+            eta: add256(block.timestamp, timelock.delay()),
+            targets: targets,
+            values: values,
+            signatures: signatures,
+            calldatas: calldatas,
+            executed: false
+        });
+
+        proposals[newProposal.id] = newProposal;
+
+        emit ProposalCreated(newProposal.id, msg.sender, targets, values, signatures, calldatas, description);
+
+        for (uint i = 0; i < newProposal.targets.length; i++) {
+          _queueOrRevert(newProposal.targets[i], newProposal.values[i], newProposal.signatures[i], newProposal.calldatas[i], newProposal.eta);
+        }
+
+        emit ProposalQueued(newProposal.id, newProposal.eta);
+        return newProposal.id;
+    }
+
+     function state(uint proposalId) public view returns (ProposalState) {
+        require(proposalCount >= proposalId && proposalId > 0, "Governor::state: invalid proposal id");
+        Proposal storage proposal = proposals[proposalId];
+        if (proposal.executed) {
+            return ProposalState.Executed;
+        } else if (block.timestamp >= add256(proposal.eta, timelock.GRACE_PERIOD())) {
+            return ProposalState.Expired;
+        } else {
+            return ProposalState.Queued;
+        }
+    }
+
+    function _queueOrRevert(address target, uint value, string memory signature, bytes memory data, uint eta) internal {
+        require(!timelock.queuedTransactions(keccak256(abi.encode(target, value, signature, data, eta))), "Governor::_queueOrRevert: proposal action already queued at eta");
+        timelock.queueTransaction(target, value, signature, data, eta);
+    }
+
+    function execute(uint proposalId) public payable {
+        require(state(proposalId) == ProposalState.Queued, "Governor::execute: proposal can only be executed if it is queued");
+        Proposal storage proposal = proposals[proposalId];
+        proposal.executed = true;
+        for (uint i = 0; i < proposal.targets.length; i++) {
+            timelock.executeTransaction.value(proposal.values[i])(proposal.targets[i], proposal.values[i], proposal.signatures[i], proposal.calldatas[i], proposal.eta);
```

Comment:
should we check the return value of the executeTransaction function call?

Reply:
executeTransaction in the timelock checks the return value. If the timelock is compromise there're a lot more problems. Hence you can't change the timelock.